File: src/main/java/me/santio/uhc/listeners/JoinListener.java
package me.santio.uhc.listeners;

import me.santio.uhc.Game;
import me.santio.uhc.models.GamePlayer;
import me.santio.uhc.sidebars.LobbySidebar;
import me.santio.uhc.states.PlayerState;
import me.santio.uhc.utils.FreezeUtils;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;

public class JoinListener implements Listener {
    
    @EventHandler
    public void onJoin(PlayerJoinEvent event) {
        GamePlayer gamePlayer = new GamePlayer(event.getPlayer().getUniqueId());
        gamePlayer.setPlayerState(Game.isStarted() ? PlayerState.DEAD : PlayerState.LOBBY);
        gamePlayer.setScoreboardState(gamePlayer.getPlayerState().getScoreboard());
        gamePlayer.getScoreboardState().getSidebar().apply(event.getPlayer());
        Game.getPlayers().put(event.getPlayer().getUniqueId(), gamePlayer);
    }
    
    @EventHandler
    public void onQuit(PlayerQuitEvent event) {
        Game.getPlayers().remove(event.getPlayer().getUniqueId());
    }
    
}

File: src/main/kotlin/me/santio/uhc/listeners/JoinListener.kt
package me.santio.uhc.listeners

import me.santio.uhc.Game
import me.santio.uhc.models.GamePlayer
import me.santio.uhc.states.PlayerState
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent

object JoinListener : Listener {
    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        val gamePlayer = GamePlayer(event.player.uniqueId)
        gamePlayer.playerState =
            if (Game.started) PlayerState.DEAD else PlayerState.LOBBY
        gamePlayer.scoreboardState = gamePlayer.playerState.scoreboard
        gamePlayer.scoreboardState.sidebar?.apply(event.player)
        Game.players[event.player.uniqueId] = gamePlayer
    }

    @EventHandler
    fun onQuit(event: PlayerQuitEvent) {
        Game.players.remove(event.player.uniqueId)
    }
}